random/src/main/java/io/airlift/random/WeightedRandomSelector.java

```diff
@@ -0,0 +1,56 @@
+package io.airlift.random;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.util.Collection;
+import java.util.function.DoubleSupplier;
+import java.util.function.ToDoubleFunction;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class WeightedRandomSelector<T>
+        implements RandomSelector<T>
+{
+    private final DoubleSupplier nextRandomDouble;
+    private final Object[] objects;
```

Comment:
`List<T>` would allow not to have unchecked casts

Reply:
Yeah - I was cautious about performance. Probably not a big deal. Changed to ArrayList.